### PR TITLE
fix(sdk): implement timeout for file_stream and add debug logs

### PR DESCRIPTION
**Pull request by kptkin**
Fixes
-----
- Fixes WB-14174

Description
-----------

### <samp>🤖 Generated by Copilot at 175974b</samp>

This pull request adds a timeout option for the file stream API, which is used to send files to the backend server. It also improves the code quality and logging of the file stream and sender modules. It affects the files `file_stream.py`, `sender.py`, `wandb_settings.py`, `progress.py`, and `settings_static.py`.

Testing
-------
How was this PR tested?

Tested in #5813

Checklist
-------
- [x] Include reference to internal ticket "Fixes WB-NNNNN" and/or GitHub issue "Fixes #NNNN" (if applicable)
- [x] Ensure PR title compliance with the [conventional commits standards](https://github.com/wandb/wandb/blob/main/CONTRIBUTING.md#conventional-commits)

### <samp>🤖 Generated by Copilot at 175974b</samp>

> _Sing, O Muse, of the skillful sender of files_
> _Who defied the limits of time and space_
> _And with cunning code and clever settings_
> _Made the file stream swift and stable._

### CODE REVIEW

File: wandb/sdk/wandb_settings.py [hunk 1]
```diff
@@ -656,6 +657,10 @@ def _default_props(self) -> Dict[str, Dict[str, Any]]:
                 ),
                 "auto_hook": True,
             },
+            _file_stream_timeout_seconds={
+                "value": 5,
```
Comment (kptkin) [inline wandb/sdk/wandb_settings.py:661]:
```suggestion
                "value": 30,
```

File: wandb/sdk/wandb_settings.py [hunk 2]
```diff
@@ -656,6 +657,10 @@ def _default_props(self) -> Dict[str, Dict[str, Any]]:
                 ),
                 "auto_hook": True,
             },
+            _file_stream_timeout_seconds={
+                "value": 30,
```
Comment (dmitryduev) [inline wandb/sdk/wandb_settings.py:661]: 30s feels risky, let's bump to to at least 60.

```suggestion
                "value": 60,
```

**Review by dmitryduev — APPROVED** [review]

**Review by dmitryduev — APPROVED** [review]
🧹 🚢

**Pull request merged**